### ISSUE TITLE
Saving Checkpoint

### ISSUE THREAD

**Issue by PaulForInvent** [UKPLab/sentence-transformers#783]
Hey,

is it possible to save a small light Checkpoint, and load this to make prediction with a trained model? Saving the whole model is very large, so I wonder if there is possibility for a more lightweight solution?

**Comment by nreimers** [comment]
This is not supported. Currently the full model is fine-tuned => the full model must be saved

**Comment by hyusterr** [comment]
I'm wondering which part of the model is saved in the output path.
in my understanding, a SBERT model contains 3 parts: transformer, pooling layer, and loss layer.
in the output path, it only contains 0_Transformer  and 1_Pooling directories.
So I'm wondering where is the loss layer saved? 
When I initialize the checkpoint in the 0_Transformer, I'm initialize only the transformer or also the loss layer?

**Comment by hyusterr** [comment]
When I want to use a checkpoint trained by triplet and apply softmax loss training, how should I initialize the checkpoint?

**Comment by nreimers** [comment]
Only the transformer + pooling. The loss layer has typically no weights and is only relevant for training.

Softmax Loss is quite bad and should be avoided when training embedding models

**Comment by hyusterr** [comment]
thanks for your instant reply!
why would you say softmax loss is quite bad? I'm wondering how did you get this observation from your experience or is there some citation?
this is insightful to me. thanks in advanced.

**Comment by nreimers** [comment]
There is no intuition why SoftMaxLoss should work. It is more a coincident. 

When training on NLI data:
https://www.sbert.net/examples/training/nli/README.html

And comparing SoftMaxLoss vs MultipleNegativesRankingLoss, models trained with MultipleNegativesRankingLoss work a lot better.

**Comment by hyusterr** [comment]
Thanks for your information!
